include/engine/engine_config.hpp

```diff
@@ -90,6 +90,7 @@ struct EngineConfig final
     int max_alternatives = 3; // set an arbitrary upper bound; can be adjusted by user
     bool use_shared_memory = true;
     Algorithm algorithm = Algorithm::CH;
+    int use_threads_number = 1;
```

Comment:
Nitpick: `use_threads_number` is kind of a weird phrasing. `number_of_threads` or just `threads`.

We should also document a value that would make TBB use the default number of threads (I think `-1`?).